### ISSUE TITLE
polyloss not working when target contains `ignore_index`

### ISSUE THREAD

**Issue by cassie101** [frgfm/Holocron#211]
### Bug description

polyloss not working when target contains `ignore_index`

### Code snippet to reproduce the bug

```
labels = [-100,2,1]
logits = [[-2.0605, -1.0522,  1.0922],[1.0303, -2.0048,  1.0727],[-1.1031,  1.0414, -2.0464]]
labels_pt = torch.tensor(labels,dtype=torch.int64)
logits_pt = torch.tensor(logits,dtype=torch.float32)

loss = poly_loss(logits_pt,labels_pt,reduction='none')

print('loss',loss)
```

### Error traceback

index -100 is out of bounds for dimension 0 with size 3

### Environment

-

**Comment by frgfm** [comment]
Hello @cassie101 👋 

Thanks for the feedback! I'll try to reproduce this on my end, would you mind sharing your environment details please? 🙏 
You can run this to display it:
```python
wget https://raw.githubusercontent.com/frgfm/Holocron/master/scripts/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

If you encounter other problems, please try to use the "bug report" template, it's much easier for me to reproduce and debug :)

Cheers ✌️

**Comment by frgfm** [comment]
Any update @cassie101 ? 🙏

**Comment by frgfm** [comment]
@cassie101 without news from you, I cannot help you :/

My best guess is that you forgot, in your snippet, to pass `ignore_index=-100`. Because without it, the loss won't handle negative values. I'm closing this for now, if you think the problem is still ongoing, feel free to reopen ✌️
